package.json

```diff
@@ -849,6 +965,23 @@
           "description": "If enabled, dragging with the mouse activates Visual mode.",
```

Comment:
nit: I was confused when I was looking at the config. The description should probably be updated since this is depended on `vim.selectmode`. Or maybe remove completely since we have selection mode?

Reply:
I was thinking of making it "deprecated" and while it is deprecated it would change the defaults of 'selectmode' to be:
```jsonc
"vim.selectmode": "key", // <- it removes the 'mouse' option so it goes to Visual Mode instead of Select Mode
```

Reply:
Actually this isn't correct. Because this setting is the equivalent of having at least `mouse=nv` or `mouse=a`. Because this is the setting that allows a mouse selection to start visual mode. The `selectmode=mouse` makes the mouse selection start Select mode. So in order to replace that setting we would have to implement the `mouse` setting. The problem is that in VIM if you have `mouse=` the mouse doesn't do anything, it doesn't even move the cursor, and even though this might be possible to do, it probably won't look good, since you'll have the cursor changing place and then go back again to where it was...

Reply:
Basically this setting on/off is the equivalent to having `on` -> `mouse=a` or `off` -> `mouse=ni`, except that when `off` the vscode selection is still kept, even though VSCodeVim doesn't know about it...

Reply:
> `on` -> `mouse=a` or `off` -> `mouse=ni` the vscode selection is still kept, even though VSCodeVim doesn't know about it

until user clicks `<shift+right>` for example. As long as we are not implementing `mouse` option, I think we just need to clarify what `vim.mouseSelectionGoesIntoVisualMode` does? Also, I don't know how hard it is to implement `mouse` option?

Since it doesn't work exactly like the `mouse` option, maybe updating the description to something like following would help?
> If enabled, dragging with the mouse activates Visual/Select mode depending on `vim.selectmode`.

Reply:
Yes the description needs to be updated. And so does the README file, I think we need to create a "Selections" section on the README to explain all things related to selections, but I really didn't want to add to the already big confusion that is the README file. We need to create an organized Wiki...

Reply:
https://github.com/VSCodeVim/Vim/wiki/Selections

If you feel so inclined, dump stuff here - even if it's not polished, better than the current state of documentation.